### PR TITLE
chore(deps): replace deprecated lib to alternative

### PR DESCRIPTION
**Pull request by yamatatsu**
## Description of your changes

This PR remove deprecated library `@aws-sdk/util-base64-node` replacing to `@smithy/util-base64` that is official alternative library.
See, https://www.npmjs.com/package/@aws-sdk/util-base64-node.

### Related issues, RFCs

**Issue number: #2246** 

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda/typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [ ] I have *added tests* that prove my change is effective and works
- [ ] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Hi, thank you for taking the time to open this PR.

As I mentioned in the linked issue, we would like to go in a different direction and instead adopt the `fromBase64` utility that we recently added to the commons package.

If you'd like to continue working on this and see the PR merged I'd kindly ask you to make changes in that regard and remove the `@smithy` dependency.

Apologies if this might cause you some rework, to avoid this type of occurrence in the future I'd advise to wait for a feedback from a maintainer in the issue before opening the PR, that way we'll be aligned from the beginning and we have a better chance to merge the PR with minor adjustments or as is.

**Comment by dreamorosi** [comment]
Hi @yamatatsu, are you still interested in continuing this PR? 

If not there's no problem at all and we'll take on the task at some point in the near future.

Please let me know, or we'll close the PR in a week.

**Comment by yamatatsu** [comment]
@dreamorosi 
Thank you for feedback. And so sorry for late response. 🙇🏻 
I've addressed your comments.
In my thinking, we can depend on `@smithy/xxx` only as devDependencies. Or we should implement `toBase64()` like as `fromBase64()`? WDYT? 🤔

**Comment by dreamorosi** [comment]
Hi @yamatatsu, no problem and thanks for coming back to this.

I agree with you, we can depend on it as a `devDependency` for now since we use `toBase64()` only in the tests.

Thanks!

File: packages/parameters/src/base/transformValue.ts
Comment (dreamorosi) [inline packages/parameters/src/base/transformValue.ts:1]: Line 70 needs to be changed to: `return new TextDecoder('utf-8').decode(fromBase64(value, 'base64'));`, the new implementation requires to specify the encoding.

Once that's done the tests will pass.

**Review by dreamorosi — CHANGES_REQUESTED** [review]
I have left one comment to fix the implementation and make the tests passing, once that's addressed we can merge this!

File: packages/parameters/src/base/transformValue.ts
Comment (yamatatsu) [reply packages/parameters/src/base/transformValue.ts:1]: Oh... Sorry that I missed that the tests failed.
Thank you for debugging it! 🙇🏻

**Review by dreamorosi — APPROVED** [review]
Thanks a lot for working on this PR and following through with all the feedback!

Congrats 🎉

**Pull request merged**
